packages/core/src/utils/cloudflare/index.ts

```diff
@@ -71,3 +71,26 @@ export const getCustomHostname = async (auth: HostnameProviderData, identifier:
 
   return result.data;
 };
+
+export const deleteCustomHostname = async (auth: HostnameProviderData, identifier: string) => {
+  const {
+    EnvSet: {
+      values: { isIntegrationTest },
+    },
+  } = await import('#src/env-set/index.js');
+  if (isIntegrationTest) {
+    return;
+  }
+
+  await got.delete(
+    new URL(
+      path.join(baseUrl.pathname, `/zones/${auth.zoneId}/custom_hostnames/${identifier}`),
+      baseUrl
+    ),
+    {
+      headers: {
+        Authorization: `Bearer ${auth.apiToken}`,
+      },
+    }
+  );
+};
```

Comment:
What happens if this method fails?

Reply:
The same as "post", there is LOG-6247 to handle more detailed error.